Inconsistent version file compared to other frameworks

A change was made in https://github.com/dotnet/aspnetcore/pull/20166 that renamed the `.version` file to `Microsoft.AspNetCore.App.versions.txt`.  It's not clear why this change was made but now there's inconsistency between ASP.NET and the other frameworks.  Microsoft.NETCore.App, Microsoft.WindowsDesktop.App and the SDK all have a file named `.version`.

Repro steps:
1. Open https://dotnetcli.blob.core.windows.net/dotnet/Sdk/5.0.100-preview.4.20223.12/dotnet-sdk-5.0.100-preview.4.20223.12-win-x64.zip
2. See the following files:
   * shared\Microsoft.AspNetCore.App\5.0.0-preview.4.20221.8\Microsoft.AspNetCore.App.versions.txt
   * shared\Microsoft.NETCore.App\5.0.0-preview.4.20221.7\\.version
   * shared\Microsoft.WindowsDesktop.App\5.0.0-preview.4.20222.5\\.version
   * sdk\5.0.100-preview.4.20223.1\\.version

There should be consistent naming amongst all the version files to make it easier for automation that consumes these files.  This name change was discovered because it broke the automation at https://github.com/dotnet/dotnet-docker/blob/fe5cedf3f4034a263fa2f118b606233cb708b1b4/eng/get-drop-versions.sh#L20.

And the reason for the name change is that Nuget disallows packing files whose names start with a dot:

> error NU5119: File 'C:\code\aspnet\aspnetcore\artifacts\obj\Microsoft.AspNetCore.App.Runtime\Debug.version' was not added to the package. Files and folders starting with '.' or ending with '.nupkg' are excluded by default. To include this file, use -NoDefaultExcludes from the commandline

--- a/src/Framework/src/Microsoft.AspNetCore.App.Runtime.csproj
+++ b/src/Framework/src/Microsoft.AspNetCore.App.Runtime.csproj
@@ -67,7 +67,8 @@ This package is an internal implementation of the .NET Core SDK and is not meant
     <!-- There is no way to suppress the .dev.runtimeconfig.json generation. -->
     <ProjectRuntimeConfigDevFilePath>$(IntermediateOutputPath)ignoreme.dev.runtimeconfig.json</ProjectRuntimeConfigDevFilePath>
 
-    <VersionFileIntermediateOutputPath>$(IntermediateOutputPath)$(SharedFxName).versions.txt</VersionFileIntermediateOutputPath>
+    <VersionTxtFileIntermediateOutputPath>$(IntermediateOutputPath)$(SharedFxName).versions.txt</VersionTxtFileIntermediateOutputPath>
+    <DotVersionFileIntermediateOutputPath>$(IntermediateOutputPath).version</DotVersionFileIntermediateOutputPath>
 
     <!-- The project representing the shared framework doesn't produce a .NET assembly or symbols. -->
     <DebugType>none</DebugType>
@@ -259,7 +260,12 @@ This package is an internal implementation of the .NET Core SDK and is not meant
     </ItemGroup>
 
     <WriteLinesToFile
-      File="$(VersionFileIntermediateOutputPath)"
+      File="$(VersionTxtFileIntermediateOutputPath)"
+      Lines="@(VersionLines)"
+      Overwrite="true" />
+
+    <WriteLinesToFile
+      File="$(DotVersionFileIntermediateOutputPath)"
       Lines="@(VersionLines)"
       Overwrite="true" />
   </Target>
@@ -402,7 +408,7 @@ This package is an internal implementation of the .NET Core SDK and is not meant
 
   <Target Name="_ResolveSharedFrameworkContent" DependsOnTargets="ResolveReferences;Crossgen">
     <ItemGroup>
-      <SharedFxContent Include="$(VersionFileIntermediateOutputPath)" />
+      <SharedFxContent Include="$(DotVersionFileIntermediateOutputPath)" />
       <SharedFxContent Include="$(ProjectDepsFilePath)" />
       <SharedFxContent Include="$(ProjectRuntimeConfigFilePath)" />
       <SharedFxContent Include="@(ReferenceCopyLocalPaths)" Condition="'%(Extension)' != '.pdb'" />
@@ -501,7 +507,7 @@ This package is an internal implementation of the .NET Core SDK and is not meant
           BeforeTargets="_GetPackageFiles">
 
     <ItemGroup>
-      <None Include="$(VersionFileIntermediateOutputPath)" Pack="true" PackagePath="." />
+      <None Include="$(VersionTxtFileIntermediateOutputPath)" Pack="true" PackagePath="." />
     </ItemGroup>
   </Target>
 

--- a/src/Framework/test/SharedFxTests.cs
+++ b/src/Framework/test/SharedFxTests.cs
@@ -133,7 +133,7 @@ namespace Microsoft.AspNetCore
         [Fact]
         public void ItContainsVersionFile()
         {
-            var versionFile = Path.Combine(_sharedFxRoot, "Microsoft.AspNetCore.App.versions.txt");
+            var versionFile = Path.Combine(_sharedFxRoot, ".version");
             AssertEx.FileExists(versionFile);
             var lines = File.ReadAllLines(versionFile);
             Assert.Equal(2, lines.Length);

--- a/src/Framework/test/SharedFxTests.cs
+++ b/src/Framework/test/SharedFxTests.cs
@@ -15,6 +15,7 @@ namespace Microsoft.AspNetCore
     {
         private readonly string _expectedTfm;
         private readonly string _expectedRid;
+        private readonly string _expectedVersionFileName;
         private readonly string _sharedFxRoot;
         private readonly ITestOutputHelper _output;
 
@@ -26,6 +27,7 @@ namespace Microsoft.AspNetCore
             _sharedFxRoot = string.IsNullOrEmpty(Environment.GetEnvironmentVariable("ASPNET_RUNTIME_PATH"))
                 ? Path.Combine(TestData.GetTestDataValue("SharedFrameworkLayoutRoot"), "shared", TestData.GetTestDataValue("RuntimePackageVersion"))
                 : Environment.GetEnvironmentVariable("ASPNET_RUNTIME_PATH");
+            _expectedVersionFileName = string.IsNullOrEmpty(Environment.GetEnvironmentVariable("ASPNET_RUNTIME_PATH")) ? ".version" : "Microsoft.AspNetCore.App.versions.txt";
         }
 
         [Fact]
@@ -133,7 +135,7 @@ namespace Microsoft.AspNetCore
         [Fact]
         public void ItContainsVersionFile()
         {
-            var versionFile = Path.Combine(_sharedFxRoot, ".version");
+            var versionFile = Path.Combine(_sharedFxRoot, _expectedVersionFileName);
             AssertEx.FileExists(versionFile);
             var lines = File.ReadAllLines(versionFile);
             Assert.Equal(2, lines.Length);

